connectors/llm/interface.py

```diff
@@ -44,15 +45,31 @@ def ask(self, system_prompt, previous_messages, question, agent_id, stream):
         prompt_params = {"context": context_text, "question": question}
         log.debug("search result: %s", context_text)
 
+        # If tiktoken doesn't support our model, default to gpt2
+        try:
+            text_splitter = tiktoken.encoding_for_model(cfg.LLM_MODEL_NAME)
```

Comment:
The LLM_MODEL_NAME can actually be an arbitrary name. It's an identifier used on the server side. What I mean is, the model might be accessed using the name `mistral-7b-instruct` on the requests made to the hosting server, but the model is actually `Mistral-7B-Instruct-v0.3`

So I think you'll want another config variable like `cfg.TOKENIZER_MODEL_NAME`

Reply:
>  when we trim the messages we need to make sure we keep the initial SystemMessage in the list because that is instructing the model on how to behave

We're not really "trimming" as much as we're "selectively not adding".  This nuance matters because the selection (token length calculation) happens after we add the default prompt here: https://github.com/RedHatInsights/tangerine-backend/pull/54/files#diff-abbf9cb2997932bbf240cd1e9f186f47e5c4c6cc15305f52aece62baa3e0fed1R56 .  So if there's anything _else_ we want to make sure we keep, we can add it before the `previous_messages` loop as well